### PR TITLE
ENH: Add the possibility to let unclose trades at end of bt.run (#273) (#343)

### PR DESCRIPTION
**Pull request by Benouare**
Hi, 
Just saw that this could be interessting for (#343).
Hope it will help. 

Cheers.

### CODE REVIEW

**Comment by mdavidg** [comment]
> Hi,
> Just saw that this could be interessting for (#343).
> Hope it will help.
> 
> Cheers.

@zlpatel  Please accept this change as soon as posible!

**Comment by zlpatel** [comment]
@mdavidg I don't have permission to do that. It would be for @kernc to do.

**Review by zlpatel — APPROVED** [review]
Looks good to me.

File: backtesting/backtesting.py [hunk 1]
```diff
@@ -984,7 +984,8 @@ def __init__(self,
                  margin: float = 1.,
                  trade_on_close=False,
                  hedging=False,
-                 exclusive_orders=False
+                 exclusive_orders=False,
+                 close_all_at_end=True
```
Comment (kernc) [inline backtesting/backtesting.py:988]: The type and amount of feedback following https://github.com/kernc/backtesting.py/issues/168 makes me think we should flip the default back to False.

Comment (kernc) [reply backtesting/backtesting.py:988]: Missing a descriptive docstring below.

File: backtesting/backtesting.py [hunk 2]
```diff
@@ -1078,7 +1079,7 @@ def __init__(self,
             warnings.warn('Data index is not datetime. Assuming simple periods, '
                           'but `pd.DateTimeIndex` is advised.',
                           stacklevel=2)
-
+        self._close_all_at_end = close_all_at_end
```
Comment (kernc) [inline backtesting/backtesting.py:1082]: Cast to `bool()` so passing `=1` is supported.

File: backtesting/test/_test.py [hunk 3]
```diff
@@ -398,6 +398,31 @@ def next(self):
...
+
+    def test_dont_close_orders_trades_from_last_strategy_iteration(self):
+        class S(Strategy):
+            def init(self): pass
+
+            def next(self):
+                if not self.position:
+                    self.buy()
+
+        self.assertGreater(len(
```
Comment (kernc) [inline backtesting/test/_test.py:423]: Can this be `assertEqual` too?

**Review by kernc — COMMENTED** [review]
Tremendous apologies for the late review. I'll try to catch up. :sweat_smile: 

Would you also add a [changelog entry](https://github.com/kernc/backtesting.py/blob/master/CHANGELOG.md)?

File: backtesting/backtesting.py (hunk 2 above)
Comment (Benouare) [reply backtesting/backtesting.py]: Done.

File: backtesting/test/_test.py (hunk 3 above)
Comment (Benouare) [reply backtesting/test/_test.py]: Done

File: backtesting/backtesting.py (hunk 1 above)
Comment (Benouare) [reply backtesting/backtesting.py]: Done.

**Comment by Benouare** [comment]
> Tremendous apologies for the late review. I'll try to catch up. 😅
> 
> Would you also add a [changelog entry](https://github.com/kernc/backtesting.py/blob/master/CHANGELOG.md)?

Done.

**Comment by Benouare** [comment]
Move to #393

**Pull request closed without merging**